packages/@lwc/compiler/src/transformers/__tests__/transform-html.spec.ts

```diff
@@ -5,27 +5,67 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 import { TransformOptions } from '../../options';
-import { transform } from '../transformer';
+import { transformSync } from '../transformer';
 
 const TRANSFORMATION_OPTIONS: TransformOptions = {
     namespace: 'x',
     name: 'foo',
 };
 
-it('should throw when processing an invalid HTML file', async () => {
-    await expect(transform(`<html`, 'foo.html', TRANSFORMATION_OPTIONS)).rejects.toMatchObject({
-        filename: 'foo.html',
-        message: expect.stringContaining('Invalid HTML syntax: eof-in-tag.'),
+describe('transformSync', () => {
+    it('should throw when processing an invalid HTML file', () => {
+        expect(() => transformSync(`<html`, 'foo.html', TRANSFORMATION_OPTIONS)).toThrow(
+            'Invalid HTML syntax: eof-in-tag.'
+        );
     });
-});
 
-it('should apply transformation for template file', async () => {
-    const actual = `
-        <template>
-            <div>Hello</div>
-        </template>
-    `;
-    const { code } = await transform(actual, 'foo.html', TRANSFORMATION_OPTIONS);
+    it('should apply transformation for template file', () => {
+        const template = `
+            <template>
+                <div>Hello</div>
+            </template>
+        `;
+        const { code } = transformSync(template, 'foo.html', TRANSFORMATION_OPTIONS);
+
+        expect(code).toContain(`tmpl.stylesheets = []`);
+    });
+
+    it('should hoist static vnodes when disableStaticContentOptimization is set to false', () => {
+        const template = `
+            <template>
+                <img src="http://www.example.com/image.png" crossorigin="anonymous">
+            </template>
+        `;
+        const { code } = transformSync(template, 'foo.html', {
+            disableStaticContentOptimization: false,
+            ...TRANSFORMATION_OPTIONS,
+        });
+
+        expect(code).toMatch('parseFragment');
```

Comment:
The `parseFragment` name might change in the future. A more resilient approach to verify this would be to check if the generated code contains the `<img` tag.

```suggestion
        expect(code).toMatch('<img');
```